src/mlpack/methods/ann/layer/multihead_attention_impl.hpp

```diff
@@ -145,30 +141,45 @@ Forward(const InputType& input, OutputType& output)
 
   // Apply the attention mask if provided. The attention mask is used to black-
   // out future sequences and generally used in Encoder-Decoder attention.
-  // The attention mask has elements 0 or -infinity.
+  // The attention mask has elements 0 or 1.
   // The shape of the attention mask : (tgtSeqLen, srcSeqLen).
   if (!attnMask.is_empty())
   {
     if (attnMask.n_rows != tgtSeqLen || attnMask.n_cols != srcSeqLen)
       Log::Fatal << "The size of the 'attn_mask' is not correct.\n";
-    scores.each_slice() += attnMask;
+    // not sure if there is a better way to do this.  now that the mask
+    // is 0 or 1, we can't simply add, but converting first seems slow
+    const arma::uword rows = tgtSeqLen;
+    const arma::uword cols = srcSeqLen;
+    for (arma::uword i=0; i<rows; i++) {
+      for (arma::uword j=0; j<cols; j++) {
+        if (attnMask.at(i, j) == 0) {
+          scores.tube(i, j).fill(-INFINITY);
```

Comment:
While this works, using macros (INFINITY) should generally be avoided if there is better functionality available.  There are two options:
* [arma::datum::inf](https://arma.sourceforge.net/docs.html#constants)
* [https://en.cppreference.com/w/cpp/types/numeric_limits/infinity](std::numeric_limits<double>::infinity())

Reply:
Yeh, thanks.  Turns out we aren't doing it this way anymore, but will keep that in mind.